Match Functionality

Silver-8ce opened the pull request:
### Purpose
To make matches function like matches

### Notes:
first PR, hope it works

closes #7716 

### Changelog:

CL: [New] You can light matches with a matchbox or shoes.

CL: [New] Matches act as lighters when lit.

CL: [New] Matches drop a burnt out match when finished burning.

Silver-8ce committed:
Matches Work

-You can light matches with matchbox, shoes.
-Matches act as lighters.
-Matches turn into burnt matchsticks.

corp-0 commented on UnityProject/Assets/Scripts/Items/Food/Torch.cs:
```diff
@@ -0,0 +1,146 @@
+		[Tooltip("Time after torch will destroy and spawn burnt remains")]
+		private float burnTimeSeconds = 30;
+
+		public SpriteHandler spriteHandler;
```
make it private and get the spriteHandler in ``Awake`` unless the prefab that will use it has more than one ``SpriteHandler``

corp-0 commented on UnityProject/Assets/Scripts/Items/Food/Torch.cs:
```diff
@@ -0,0 +1,146 @@
+    /// <summary>
+    /// Lightable objects that can be used as lighters
+    /// </summary>
+    public class Torch : NetworkBehaviour, ICheckedInteractable<HandApply>,
```
why inside the ``Food`` folder?

Silver-8ce replied:
Because that is the location of both lighters and cigarettes

corp-0 replied:
fair

corp-0 commented on UnityProject/Assets/Scripts/Items/Food/Torch.cs:
```diff
@@ -0,0 +1,146 @@
+		public SpriteHandler spriteHandler;
+		private FireSource fireSource;
+		private Pickupable pickupable;
+		private ItemLightControl lightControl;
```
seems like you forgot to use this

Silver-8ce replied:
thanks for the help

Silver-8ce committed:
Update Torch.cs

Silver-8ce committed:
Codacy Changes

I don't understand why it would want 'LightableSurface' to be private in CommonTraits.

Silver-8ce committed:
Random ID change

hopefully this works

corp-0 commented on UnityProject/Assets/Scripts/Items/Food/Torch.cs:
```diff
@@ -0,0 +1,143 @@
+				ServerChangeLit(true);
+			}
+
+			return;
```
no need to return at the end of a void method. BTW the name of the method implies it will make some kind of validation or that the action might fail, change it to  ``LightByObject``.

NoooneyDude commented on UnityProject/Assets/ScriptableObjects/Traits/CommonTraitsSingleton.asset:
```diff
@@ -63,6 +63,7 @@ MonoBehaviour:
   Loomable: {fileID: 11400000, guid: 239e6bd28c3fc4442bf2dc5420eada94, type: 2}
   CanisterFillable: {fileID: 11400000, guid: a88f1555ca8abf643aea180c637f5329, type: 2}
   Breakable: {fileID: 11400000, guid: 86782de91b339c1438d9f4e310d321f4, type: 2}
+  LightableSurface: {fileID: 11400000, guid: 7410b4f6e26235e4f93aa1ebcbab0cec, type: 2}
```
I suggest serializing reference to this trait on the match script instead, as it probably won't be used much elsewhere. There's a few (a lot of) other traits that probably don't belong here too (though that isn't your problem : ) ).

Silver-8ce committed:
Cleanup

Silver-8ce committed:
Update CommonTraitsSingleton.asset

Silver-8ce committed:
Update NetworkManager.prefab

Silver-8ce committed:
Delete Asteroid05.unity

Bod9001 commented on UnityProject/Assets/Tests/NewTestScript.cs:
```diff
@@ -0,0 +1,25 @@
+{
+    // A Test behaves as an ordinary method
+    [Test]
+    public void NewTestScriptSimplePasses()
```
seems you left this in accidentally

Silver-8ce committed:
Delete Asteroid06.unity

Silver-8ce committed:
Delete Asteroid07.unity

Silver-8ce committed:
Delete Asteroid09.unity

Bod9001 commented on UnityProject/Assets/Scenes/AsteroidScenes/Asteroid05.unity:
```diff
@@ -1,6650 +0,0 @@
-  m_ObjectHideFlags: 0
-  serializedVersion: 2
-  m_OcclusionBakeSettings:
-    smallestOccluder: 5
```
why deleting?

Silver-8ce replied:
Well, I accidentally added an additional 21 unintended changes

Silver-8ce replied:
And I'm not exactly sure how to revert those change

Silver-8ce committed:
Revert "Update NetworkManager.prefab"

This reverts commit 5a55125ff99cb0101b4ab16daa69110fda33e486.

Silver-8ce committed:
Merge branch 'BigPeteBranch' of https://github.com/Silver-8ce/unitystation into BigPeteBranch

Silver-8ce committed:
Revert "Revert "Update NetworkManager.prefab""

This reverts commit 1ac998a6be4387597f38423e637a60b053762f9f.

Silver-8ce committed:
Revert

NoooneyDude reviewed (approved):
Thanks for your contribution!

Just for future reference, our [standards guideline](https://unitystation.github.io/unitystation/contribution-guides/Development-Standards-Guide/) suggests to use an explicit false check instead of boolean negation, `bool == false` over `!bool`.
